cunumeric/deferred.py

```diff
@@ -503,6 +503,25 @@ def _create_indexing_array(
                         f"dimension {i} doesn't match to the shape of the"
                         f"index array which is {rhs.shape[i]}"
                     )
+
+            if key.size == 0:
+                rhs_store = rhs.base
+                for i in range(key.ndim - 1):
+                    rhs_store = rhs_store.project(0, 0)
```

Comment:
This will fail if one of the first `key.ndim - 2` dimensions on `rhs` is zero-size.